packages/react-scripts/template/README.md

```diff
@@ -852,6 +853,49 @@ In the future we plan to integrate it into Create React App even more closely.
 
 To learn more about Flow, check out [its documentation](https://flow.org/).
 
+## Adding TypeScript
+
+TypeScript is a typed superset of JavaScript that compiles to plain JavaScript.
+
+Recent versions of [TypeScript](https://www.typescriptlang.org/) work with Create React App projects out of the box thanks to Babel 7. Beware that Babel 7 TypeScript does not allow some features of TypeScript such as constant enum and namespaces.
+
+To add TypeScript to a Create React App project, follow these steps:
+
+1. Run `npm install --dev @types/react @types/react-dom @types/jest` (or `yarn add --dev @types/react @types/react-dom @types/jest`).
+2. Rename the `.js` files you want to convert. Use `.tsx` if they use JSX or `.ts` if not (e.g. `git mv src/index.js src/index.tsx`).
+
+3. Create a [tsconfig.json](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) file at the root directory with the following content:
+
+   ```json
+   {
+     "compilerOptions": {
+       "target": "es5",
```

Comment:
Is this field ever used when using babel to compile?

Reply:
Since Babel is being used, it won't affect the emit. The `lib` compiler option has been set so the defaults set by `target` are not used.

That said, if a user wants to add secondary build script using `ts-node`, it might be helpful to have target set to `es5`.